wagtail/models/i18n.py

```diff
@@ -210,6 +220,26 @@ def check(cls, **kwargs):
                 )
             )
 
+        if is_translation_model and not any(
+            isinstance(constraint, models.UniqueConstraint)
+            and set(constraint.fields) == {"translation_key", "locale"}
+            for constraint in constraints
```

Comment:
```suggestion
            and set(constraint.fields) == {"translation_key", "locale"}
            for constraint in cls._meta.constraints
```

Reply:
@laymonage I had this `constraints` instead of this `cls._meta.constraints` because the latter would require I define the `UniqueConstraint` in the `meta` class which would keep triggering the database migration error as you wrote below.

> The other one is `wagtail.tests.test_migrations.TestForMigrations. test__migrations`, which ensures that running `python manage.py makemigrations` will not create spurious migrations. This is because we still need to either (a) update the migrations or (b) create new migrations that drop and recreate the constraints for the test models.

For `constraints` to work, I had to define it in the `check` method as you pointed out above.

```python
constraints = [
            models.UniqueConstraint(
                fields=["translation_key", "locale"],
                name="unique_translation_key_locale_%(app_label)s_%(class)s",
            )
        ]
```

This is only serving the function of being used in for the loop `for constraint in constraints`.

Reply:
But also, I could be missing out something. Please help me clarify. Thanks so much for the assistance so far.

Reply:
@laymonage regardless of all my comments above, I have made the suggested change above. But I am aware this would cause a failing test with the migration, partcularly, `wagtail.tests.test_migrations.TestForMigrations. test__migration`, since I would now have to define `constraints` in the `meta` class . Let me know if you want me to create the necessary migration files. I haven't done that yet since my most recent changes haven't been reviewed yet. Thanks so much.